### DISCUSSION TITLE
MyST Enhancement Proposals

### DISCUSSION THREAD

**Discussion by rowanc1** [executablebooks/meta#662] [General]
I am currently going through an entire pass of MyST syntax and starting to spec out extensions in the well-typed/specified `mdast` ([Markdown AST](https://github.com/syntax-tree/mdast)) ecosystem. I think there is real promise here of starting to introduce a test suite that can be used across implementations to show adherence to a MyST Standard/Core/Spec/Test Suite. As part of this I am also improving documentation for MyST that includes live examples in the docs.

In going through this process and starting to document and compare our implementation to others (pandoc, rmarkdown, mdx, unist directives, idyll .. as well as rst/sphinx) I am writing down some of the places where I think there are improvements that we could make to MyST.

I am wondering how best to share these?

I am envisioning a few documents that explain some of the challenges/shortfalls of the current syntax, and how we can make evolution tweaks to improve/standardize over time. I am looking for something more like a long-form [PEP](https://www.python.org/dev/peps/pep-0001/) than a github discussion/issue -- a [MEP](https://acronyms.thefreedictionary.com/MEP) if you will.

@choldgraf or others do you have ideas on how best to start the process around this?

Possible way forward: create a repo (myst-enhancement-proposals) and submit the first one describing the process for input (modelling after PEPs and JEPs). I think this advances some of the discussions around governance and community laid out in https://github.com/executablebooks/meta/issues/493.

I am happy to take on some of the initial writing/documenting and submit it for review/discussion. There is some conversation in [this document](https://hackmd.io/@choldgraf/HkgAIHtQY) that we could adopt JEPs directly. I think that those are too "big" for this sort of discussion and having a designated place for myst specifically would be helpful in gaining traction in other communities, demonstrating our commitment to introducing and supporting an evolving markdown spec for computational, publication-quality documents.

**Comment by chrisjsewell** [comment]
> Possible way forward: create a repo (myst-enhancement-proposals) and submit the first one describing the process for input (modelling after PEPs and JEPs).

As mentioned, I created https://github.com/executablebooks/extension-proposals a while back
